### DISCUSSION TITLE
DuckDB python API: query composition

### DISCUSSION THREAD

**Discussion by mpage-uga** [duckdb/duckdb#2066] [Q&A] [answered]
Suppose I use DuckDB with python, for querying an Apache parquet file `test.pq` with a table containing two columns `f1` and `f2`.

`r1 = duckdb.query("""SELECT f1 FROM parquet_scan('test.pq') WHERE f2 > 1""")`

Now I would like to use `r1` result in another query, like:

`duckdb.query("""SELECT * FROM r1 WHERE f1 > 10""")`

However, last instruction gives: RuntimeError: Catalog Error: Table with name r1 does not exist!

Am I missing a DuckDB method equivalent to Apache Spark `registerTempTable()` ?

**Comment by Mytherin** [comment, marked as answer]
You could do this:

```sql
-- create a persistent/materialized table with the name r1
duckdb.query("""SELECT f1 FROM parquet_scan('test.pq') WHERE f2 > 1""").create("r1");
-- create a view with the name r1
duckdb.query("""SELECT f1 FROM parquet_scan('test.pq') WHERE f2 > 1""").create_view("r1");
```

But replacement scans in Python should be extended so that they also work on our own relational objects. This would make it so your example works as well.
